src/movies.js

```diff
@@ -1,25 +1,94 @@
 // Iteration 1: All directors? - Get the array of all directors.
+function getAllDirectors(movies){
+  const directorsArray = movies.map(film => film.director);
+  return directorsArray;
+}
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllFilteredDirectors(arr) {
+  const directorsWithDuplicates = movies.map(film => film.director);
+    const filtered = directorsWithDuplicates.filter(function(director, index){
+      return directorsWithDuplicates.indexOf(director) === index
+    })
+    return filtered
+  }
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(arr) {
+  const moviesDirectedBySteven = arr.filter(function(movie){
+      if (movie.director === 'Steven Spielberg'){
+        return movie;
+      }
+    }).filter(function(movie){
+      return movie.genre.includes('Drama')
+    });
+    return moviesDirectedBySteven.length
+    }
```

Comment:
**Iteration 2:**
Nice job. Here you can use the logical AND (&&) in your if statement and by that combine two if statements into one and with that you don't have to use the second filter().